### PR TITLE
:zap: Remove redundant dependency

### PR DESCRIPTION
**Pull request by GoogleFeud**
This is a pull request that removes the `axios` dependancy and replaces it with `node-fetch`.

The `node-fetch` library is a http client, and it's already used by discord.js to make HTTP calls to the discord API. Adding a new dependancy that does the same thing for only GET requests seems redundant. `node-fetch` can be used instead, as it's already required by discord.js.

The new code has been tested against the latest version of discord.js. 

Here's the code used to test the changes: (test/index.js)

```js
const Discord = require("discord.js");
const Backup = require("../lib/index.js");
const fs = require("fs");

const client = new Discord.Client();

client.on("message", async message => {
     if (message.content === "!create") {
           const backup = await Backup.create(message.guild, {jsonBeautify: true, saveImages: "base64"});
           console.log(backup); 
           fs.writeFileSync("./serverIcon.jpg", backup.iconBase64, {encoding: "base64"});
           message.channel.send("Created!");
     }
}); 
client.login("token");
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Guild, SnowflakeUtil, version as djsVersion } from 'discord.js';
 const master: boolean = djsVersion.split('.')[0] === '12';
 
-import axios from 'axios';
+import nodeFetch from "node-fetch";
```
Comment (Androz2091) [inline src/index.ts:4]:
```suggestion
import fetch from 'node-fetch';
```

File: src/index.ts [hunk 2]
```diff
@@ -99,22 +99,19 @@ export const create = async (
                 };
                 if (guild.iconURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.iconURL(), { responseType: 'arraybuffer' });
-                        backupData.iconBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.iconBase64 = (await nodeFetch(guild.iconURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:102]:
```suggestion
                        backupData.iconBase64 = (await fetch(guild.iconURL()).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 3]
```diff
@@ -99,22 +99,19 @@ export const create = async (
...
                     }
                     backupData.iconURL = guild.iconURL();
                 }
                 if (guild.splashURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.splashURL(), { responseType: 'arraybuffer' });
-                        backupData.splashBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.splashBase64 = (await nodeFetch(guild.splashURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:108]:
```suggestion
                        backupData.splashBase64 = (await fetch(guild.splashURL()).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 4]
```diff
@@ -99,22 +99,19 @@ export const create = async (
...
                     }
                     backupData.splashURL = guild.splashURL();
                 }
                 if (guild.bannerURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.bannerURL(), { responseType: 'arraybuffer' });
-                        backupData.bannerBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.bannerBase64 = (await nodeFetch(guild.bannerURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:114]:
```suggestion
                        backupData.bannerBase64 = (await fetch(guild.bannerURL()).then(res => res.buffer())).toString("base64");
```

File: src/master/create.ts [hunk 5]
```diff
@@ -1,5 +1,7 @@
-import axios from 'axios';
-import { CategoryChannel, Guild, TextChannel, VoiceChannel } from 'discord.js';
+
+import fetch from "node-fetch";
+
+import { CategoryChannel, Guild, TextChannel, VoiceChannel} from 'discord.js';
```
Comment (Androz2091) [inline src/master/create.ts:4]:
```suggestion
import fetch from 'node-fetch';
import { CategoryChannel, Guild, TextChannel, VoiceChannel} from 'discord.js';
```

**Comment by GoogleFeud** [comment]
Sorry, did that by accident!

**Comment by Androz2091** [comment]
No problem, but could you revert the changes so it keeps the same indent please?

**Comment by GoogleFeud** [comment]
Just pulled the suggested changes, and TS is screaming this:

`Individual declarations in merged declaration 'fetch' must be all exported or all local`
and
`Import declaration conflicts with local declaration of 'fetch'`

(Also, I think the tslint.json identation got fixed)

File: src/master/create.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
-import axios from 'axios';
-import { CategoryChannel, Guild, TextChannel, VoiceChannel } from 'discord.js';
+import fetch from 'node-fetch';
```
Comment (Androz2091) [inline src/master/create.ts:1]:
```suggestion
import nodeFetch from 'node-fetch';
```

File: src/master/create.ts [hunk 7]
```diff
@@ -69,8 +69,7 @@ export async function getEmojis(guild: Guild, options: CreateOptions) {
             name: emoji.name
         };
         if (options.saveImages && options.saveImages === 'base64') {
-            const res = await axios.get(emoji.url, { responseType: 'arraybuffer' });
-            eData.base64 = Buffer.from(res.data, 'binary').toString('base64');
+            eData.base64 = (await fetch(emoji.url).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/master/create.ts:72]:
```suggestion
            eData.base64 = (await nodeFetch(emoji.url).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 8]
```diff
@@ -99,22 +99,19 @@ export const create = async (
...
-                        backupData.splashBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.splashBase64 = (await fetch(guild.splashURL()).then(res => res.buffer())).toString("base64");
                     }
                     backupData.splashURL = guild.splashURL();
                 }
                 if (guild.bannerURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.bannerURL(), { responseType: 'arraybuffer' });
-                        backupData.bannerBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.bannerBase64 = (await fetch(guild.bannerURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:114]:
```suggestion
                        backupData.bannerBase64 = (await nodeFetch(guild.bannerURL()).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 9]
```diff
@@ -99,22 +99,19 @@ export const create = async (
...
-                        backupData.iconBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.iconBase64 = (await fetch(guild.iconURL()).then(res => res.buffer())).toString("base64");
                     }
                     backupData.iconURL = guild.iconURL();
                 }
                 if (guild.splashURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.splashURL(), { responseType: 'arraybuffer' });
-                        backupData.splashBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.splashBase64 = (await fetch(guild.splashURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:108]:
```suggestion
                        backupData.splashBase64 = (await nodeFetch(guild.splashURL()).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 10]
```diff
@@ -99,22 +99,19 @@ export const create = async (
                 };
                 if (guild.iconURL()) {
                     if (options && options.saveImages && options.saveImages === 'base64') {
-                        const res = await axios.get(guild.iconURL(), { responseType: 'arraybuffer' });
-                        backupData.iconBase64 = Buffer.from(res.data, 'binary').toString('base64');
+                        backupData.iconBase64 = (await fetch(guild.iconURL()).then(res => res.buffer())).toString("base64");
```
Comment (Androz2091) [inline src/index.ts:102]:
```suggestion
                        backupData.iconBase64 = (await nodeFetch(guild.iconURL()).then(res => res.buffer())).toString("base64");
```

File: src/index.ts [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import { Guild, SnowflakeUtil, version as djsVersion } from 'discord.js';
 const master: boolean = djsVersion.split('.')[0] === '12';
 
-import axios from 'axios';
+import fetch from 'node-fetch';
```
Comment (Androz2091) [inline src/index.ts:4]:
```suggestion
import nodeFetch from 'node-fetch';
```

**Pull request merged**